bluemira/geometry/coordinates.py

```diff
@@ -643,6 +646,35 @@ def __init__(self, xyz_array):
         self._is_planar = None
         self._normal_vector = None
 
+    @classmethod
+    def from_json(cls, filename):
+        """
+        Load a Coordinates object either from a JSON file.
+
+        Parameters
+        ----------
+        filename: str
+            Full path file name of the data
+        """
+        ext = os.path.splitext(filename)[-1]
+        if ext == "":
+            # Default to JSON in the absence of an extension
+            filename += ".json"
+
+        elif ext != ".json":
+            raise CoordinatesError(
+                f"File extension {ext} not recognised for Coordinates."
+            )
```

Comment:
This feels weird. I shouldn't have to name my file with a ".json" extension. This catch should be a try except on the `json.load` leaving the filename untouched